heron/tmaster/tests/cpp/server/stateful_restorer_unittest.cpp

```diff
@@ -0,0 +1,436 @@
+  topology->set_id(topology_id);
+  topology->set_name(topology_name);
+  size_t spouts_size = num_spouts;
+  size_t bolts_size = num_bolts;
+  // Set spouts
+  for (size_t i = 0; i < spouts_size; ++i) {
+    heron::proto::api::Spout* spout = topology->add_spouts();
+    // Set the component information
+    heron::proto::api::Component* component = spout->mutable_comp();
+    sp_string compname = SPOUT_NAME;
+    compname += std::to_string(i);
+    component->set_name(compname);
+    heron::proto::api::ComponentObjectSpec compspec = heron::proto::api::JAVA_CLASS_NAME;
+    component->set_spec(compspec);
+    // Set the stream information
+    heron::proto::api::OutputStream* ostream = spout->add_outputs();
+    heron::proto::api::StreamId* tstream = ostream->mutable_stream();
+    sp_string streamid = STREAM_NAME;
+    streamid += std::to_string(i);
+    tstream->set_id(streamid);
+    tstream->set_component_name(compname);
+    heron::proto::api::StreamSchema* schema = ostream->mutable_schema();
+    heron::proto::api::StreamSchema::KeyType* key_type = schema->add_keys();
+    key_type->set_key("dummy");
+    key_type->set_type(heron::proto::api::OBJECT);
+    // Set the config
+    heron::proto::api::Config* config = component->mutable_config();
+    heron::proto::api::Config::KeyValue* kv = config->add_kvs();
+    kv->set_key(heron::config::TopologyConfigVars::TOPOLOGY_COMPONENT_PARALLELISM);
+    kv->set_value(std::to_string(num_spout_instances));
+  }
+  // Set bolts
+  for (size_t i = 0; i < bolts_size; ++i) {
+    heron::proto::api::Bolt* bolt = topology->add_bolts();
+    // Set the component information
+    heron::proto::api::Component* component = bolt->mutable_comp();
+    sp_string compname = BOLT_NAME;
+    compname += std::to_string(i);
+    component->set_name(compname);
+    heron::proto::api::ComponentObjectSpec compspec = heron::proto::api::JAVA_CLASS_NAME;
+    component->set_spec(compspec);
+    // Set the stream information
+    heron::proto::api::InputStream* istream = bolt->add_inputs();
+    heron::proto::api::StreamId* tstream = istream->mutable_stream();
+    sp_string streamid = STREAM_NAME;
+    streamid += std::to_string(i);
+    tstream->set_id(streamid);
+    sp_string input_compname = SPOUT_NAME;
+    input_compname += std::to_string(i);
+    tstream->set_component_name(input_compname);
+    istream->set_gtype(grouping);
+    // Set the config
+    heron::proto::api::Config* config = component->mutable_config();
+    heron::proto::api::Config::KeyValue* kv = config->add_kvs();
+    kv->set_key(heron::config::TopologyConfigVars::TOPOLOGY_COMPONENT_PARALLELISM);
+    kv->set_value(std::to_string(num_bolt_instances));
+  }
+  // Set message timeout
+  heron::proto::api::Config* topology_config = topology->mutable_topology_config();
+  heron::proto::api::Config::KeyValue* kv = topology_config->add_kvs();
+  kv->set_key(heron::config::TopologyConfigVars::TOPOLOGY_RELIABILITY_MODE);
+  kv->set_value("EXACTLY_ONCE");
```

Comment:
Can you actually have
heron::config::TopologyConfigVars::EXACTLY_ONCE
instead of a "EXACTLY_ONCE"